examples/hooks-next/pages/index.tsx

```diff
@@ -83,9 +83,8 @@ function FallbackComponent({ attribute }: { attribute: string }) {
 }
 
 export async function getServerSideProps({ req }) {
-  const url = new URL(
-    req.headers.referer || `https://${req.headers.host}${req.url}`
-  ).toString();
+  const protocol = req.headers.referer?.split('://')[0] || 'https';
+  const url = `${protocol}://${req.headers.host}${req.url}`;
```

Comment:
```suggestion
  const url = new URL(
      req.headers.referer || `https://${req.headers.host}${req.url}`, 'https://example.com'
    ).toString();
```

Does this allow the right kind of fallback without us having to concatenate the url completely?

Reply:
`req.headers.referer` is not correct here because it always remains the same URL.